body.tex

```diff
@@ -295,8 +303,13 @@ \subsection{Registering}
 \validation
 \begin{itemize}
     \item $\field{P}{org}$ identifies an existing org $O$,
-    \item $\field{P}{id}$ is unique under $O$, between $1$ and $32$
-        bytes long, and valid \textsf{UTF-8}.
+    \item $\field{P}{id}$ is unique under $O$, and complies with the
+        following rules:
+        \begin{itemize}
+            \item Must be between $1$ and $32$ bytes long.
+            \item Must only contain \texttt{a-z}, \texttt{0-9}, `\texttt{-}'
+                and `\texttt{\_}' characters.
```

Comment:
> and `\texttt{\_}' characters.

To make sure: this is accepted for project id/names but not for org identifiers?

Reply:
Yeah exactly, project names are a bit more permissive because they are never used on their own. Interestingly, github allows a repo to be called `_` or `-foo-`, which this spec would also allow. I don't really have a good reason to disallow it if the user wants.. given that it'll be prefixed with the org name in almost all scenarios anyway.